tests/e2e/deployment/deployment_test.go

```diff
@@ -57,59 +56,68 @@ var _ = Describe("Application deployment test in E2E scenario", func() {
 			// Start test timer
 			testTimer = DeploymentTestTimerGroup.NewTestTimer(testSpecReport.LeafNodeText)
 		})
+
 		AfterEach(func() {
 			// End test timer
 			testTimer.End()
 			// Print result
 			testTimer.PrintResult()
-			var podlist corev1.PodList
-			var deploymentList appsv1.DeploymentList
-			err := utils.GetDeployments(&deploymentList, ctx.Cfg.K8SMasterForKubeEdge+constants.DeploymentHandler)
+
+			By(fmt.Sprintf("get deployment %s", UID))
+			deployment, err := utils.GetDeployment(clientSet, metav1.NamespaceDefault, UID)
 			Expect(err).To(BeNil())
-			for _, deployment := range deploymentList.Items {
-				if deployment.Name == UID {
-					label := nodeName
-					podlist, err = utils.GetPods(ctx.Cfg.K8SMasterForKubeEdge+constants.AppHandler, label)
-					Expect(err).To(BeNil())
-					StatusCode := utils.DeleteDeployment(ctx.Cfg.K8SMasterForKubeEdge+constants.DeploymentHandler, deployment.Name)
-					Expect(StatusCode).Should(Equal(http.StatusOK))
-				}
-			}
-			utils.CheckPodDeleteState(ctx.Cfg.K8SMasterForKubeEdge+constants.AppHandler, podlist)
+
+			By(fmt.Sprintf("list pod for deploy %s", UID))
+			labelSelector := labels.SelectorFromSet(map[string]string{"app": UID})
+			_, err = utils.ListPods(clientSet, metav1.NamespaceDefault, labelSelector, nil)
+			Expect(err).To(BeNil())
+
+			By(fmt.Sprintf("delete deploy %s", UID))
+			err = utils.DeleteDeployment(clientSet, deployment.Namespace, deployment.Name)
+			Expect(err).To(BeNil())
+
+			By(fmt.Sprintf("wait for pod of deploy %s disappear", UID))
+			err = utils.WaitForPodsToDisappear(clientSet, metav1.NamespaceDefault, labelSelector, constants.Interval, constants.Timeout)
+			Expect(err).To(BeNil())
+
 			utils.PrintTestcaseNameandStatus()
 		})
 
 		It("E2E_APP_DEPLOYMENT_1: Create deployment and check the pods are coming up correctly", func() {
-			replica := 1
+			replica := int32(1)
```

Comment:
in kubernetes deployment API, the type of `Replicas` is `int32`, I updated the type Replicas from int to int32 in  releated functions, so we need this change, otherwise it will raise an err